### PR TITLE
enable JWT token param in frontend API calls

### PR DESCRIPTION
**Pull request by Yuqing-cat**
- Enable JWT token to replace current hardcode mock token.
- Protect all API calls with JWT token
- Abstract MsalInstance logic into `utils.tsx`

### CODE REVIEW

File: ui/src/api/api.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import Axios from "axios";
 import { DataSource, Feature, FeatureLineage, UserRole, Role } from "../models/model";
+import { InteractionRequiredAuthError, PublicClientApplication } from "@azure/msal-browser";
 import mockUserRole from "./mock/userrole.json";
+import { getMsalConfig } from "../utils/utils";
 
 const API_ENDPOINT = process.env.REACT_APP_API_ENDPOINT + "/api/v1";
-const token = "mockAppServiceKey";
+const MOCK_TOKEN = "mockAppServiceKey";
```
Comment (blrchen) [inline ui/src/api/api.tsx:8]: const MOCK_TOKEN = "mockAppServiceKey";
this should get removed

File: ui/src/api/api.tsx [hunk 2]
```diff
@@ -139,3 +153,23 @@ export const deleteUserRole = async (role: Role) => {
...
 }
+
+export const getIdToken = async( msalInstance: PublicClientApplication ): Promise<string> => {
+  const activeAccount = msalInstance.getActiveAccount(); // This will only return a non-null value if you have logic somewhere else that calls the setActiveAccount API
+  const accounts = msalInstance.getAllAccounts();
+  const request = {
+      scopes: ["User.Read"],
+      account: activeAccount || accounts[0]
+  };
+  await msalInstance.acquireTokenSilent(request).then(response => {
```
Comment (blrchen) [inline ui/src/api/api.tsx:164]: >acquireTokenSilent [](http://example.com/codeflow?start=21&length=18)

It seems acquireTokenSilent will look up token from local cache first, if not found or expired, this method will fetch / refresh token. Is this correct?

If refresh token is taken be MADL internally, it's better add a comment here.

File: ui/src/api/api.tsx [hunk 3]
```diff
@@ -139,3 +153,23 @@ export const deleteUserRole = async (role: Role) => {
...
+    console.log(response.idToken)
```
Comment (blrchen) [inline ui/src/api/api.tsx:165]: >console.log(response.idToken) [](http://example.com/codeflow?start=4&length=29)

console.log(response.idToken)
Let's remove this

**Review by blrchen — APPROVED** [review]
:shipit:

**Review by xiaoyongzhu — APPROVED** [review]

**Pull request merged**